ckb-next: 0.6.2 -> 0.6.2-unstable-2025-09-25; Switch to Qt6

4JX opened the pull request:
Updates `ckb-next` to build with Qt6, release pending. Ignores that it technically can still build with Qt5 given a few tweaks.

Fixes #444209 which is caused by #443745. `dbusmenu-qt5` is [only used on the Qt5 variant of the app](https://github.com/ckb-next/ckb-next/blob/4bf942dba5e73c2778ef797b6b8dd6b0239aca9a/src/gui/CMakeLists.txt#L336-L346) (as per the name implies).

Qt6 building is available since `0.6.1`, but there's a few fixes that [haven't made it](https://github.com/ckb-next/ckb-next/commit/1ce29099e3f79269f43b6e09003eff2dbece8048) [to a release tag yet](https://github.com/ckb-next/ckb-next/commit/8566997efabd67c7f7f31b0caa4f8cd1d29252f6).

<img width="500" alt="image" src="https://github.com/user-attachments/assets/ebd03fdb-8f83-4603-b9ae-8dffe6b94c1d" />

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [X] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

LordGrimmauld commented on pkgs/top-level/all-packages.nix:
```diff
@@ -2571,7 +2571,7 @@ with pkgs;
   cffconvert = python3Packages.toPythonApplication python3Packages.cffconvert;
 
-  ckb-next = libsForQt5.callPackage ../tools/misc/ckb-next { };
+  ckb-next = kdePackages.callPackage ../tools/misc/ckb-next { };
```
Please don't do `kdePackages.callPackage` (or `qt6.callPackage`). Instead `qt6` should be an argument to your package, using packages by explicit `qt6.qtbase` (and all the other qt packages you need). Then you can use a regular `callPackage` (or, even better, move the package to `pkgs/by-name`)

4JX replied:
Done. Used `qt6`/`qt6Packages` as I saw applicable. Also moved it to `by-name`.

<details><summary>./ci/nixpkgs-vet.sh master</summary>

```
Using HEAD commit 0d2ba1acef8545dc078e13afafb1a905f4d06b17
Creating Git worktree for the HEAD commit in /tmp/nix-shell-82865-1571987213/tmp.s1fNb6MFMA/merged.. Done
Fetching base branch master to compare against.. c467fef3524c4d74b6aeb6c0ef90329561dd09c6
Creating Git worktree for the base branch in /tmp/nix-shell-82865-1571987213/tmp.s1fNb6MFMA/base.. Done
Merging base branch into the HEAD commit in /tmp/nix-shell-82865-1571987213/tmp.s1fNb6MFMA/merged.. 87cd528e78f62064c565d6245dd9909a4bf88b28
Running nixpkgs-vet..
this derivation will be built:
  /nix/store/b131vyv04hssj412ws4z96slyr45mvnx-nixpkgs-vet.drv
these 10 paths will be fetched (5.39 MiB download, 20.92 MiB unpacked):
  /nix/store/d31b404dkz0z9jk53y5ppfq2bsvl398a-nix-2.32.2
  /nix/store/xghcfng5a1s9690ny04z4ckaqi8pyxbz-nix-2.32.2
  /nix/store/92ms5lw87m38057jznga61cs49nmzr50-nix-cmd-2.32.2
  /nix/store/9cimjqf84j94q4wpzpjh3jm9byvznzgh-nix-expr-2.32.2
  /nix/store/bgawslnidn724chg7zk95c6r373wjy9l-nix-fetchers-2.32.2
  /nix/store/qhsihscq8sb3sv37qxp7amyss3a790v7-nix-flake-2.32.2
  /nix/store/74m02mxr9isxlalpbxai2kawmg6gzps3-nix-main-2.32.2
  /nix/store/bsd4afzz8c00dksbhcs05jshydwaxx8y-nix-store-2.32.2
  /nix/store/nn5fi439n98l1gdaag8mlnrcsln2ik3b-nix-util-2.32.2
  /nix/store/gfq8nxyr5ih7bzi67kfn9swjh0hsjjix-nixpkgs-vet-0.1.4
copying path '/nix/store/gfq8nxyr5ih7bzi67kfn9swjh0hsjjix-nixpkgs-vet-0.1.4' from 'https://cache.nixos.org'...
copying path '/nix/store/nn5fi439n98l1gdaag8mlnrcsln2ik3b-nix-util-2.32.2' from 'https://cache.nixos.org'...
copying path '/nix/store/bsd4afzz8c00dksbhcs05jshydwaxx8y-nix-store-2.32.2' from 'https://cache.nixos.org'...
copying path '/nix/store/bgawslnidn724chg7zk95c6r373wjy9l-nix-fetchers-2.32.2' from 'https://cache.nixos.org'...
copying path '/nix/store/74m02mxr9isxlalpbxai2kawmg6gzps3-nix-main-2.32.2' from 'https://cache.nixos.org'...
copying path '/nix/store/9cimjqf84j94q4wpzpjh3jm9byvznzgh-nix-expr-2.32.2' from 'https://cache.nixos.org'...
copying path '/nix/store/qhsihscq8sb3sv37qxp7amyss3a790v7-nix-flake-2.32.2' from 'https://cache.nixos.org'...
copying path '/nix/store/92ms5lw87m38057jznga61cs49nmzr50-nix-cmd-2.32.2' from 'https://cache.nixos.org'...
copying path '/nix/store/xghcfng5a1s9690ny04z4ckaqi8pyxbz-nix-2.32.2' from 'https://cache.nixos.org'...
copying path '/nix/store/d31b404dkz0z9jk53y5ppfq2bsvl398a-nix-2.32.2' from 'https://cache.nixos.org'...
building '/nix/store/b131vyv04hssj412ws4z96slyr45mvnx-nixpkgs-vet.drv'...
Validated successfully
/nix/store/hwx3frzv73j63ki9cq0d5vhpdzjj3l76-nixpkgs-vet
Cleaning up.. Done
```
</details>

<details><summary>nixpkgs-review rev HEAD</summary>

```
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0
From https://github.com/NixOS/nixpkgs
   9c9d1a889480..c467fef3524c  master     -> refs/nixpkgs-review/0
$ git worktree add /home/infinity/.cache/nixpkgs-review/rev-0d2ba1acef8545dc078e13afafb1a905f4d06b17/nixpkgs c467fef3524c4d74b6aeb6c0ef90329561dd09c6
Preparing worktree (detached HEAD c467fef3524c)
Updating files: 100% (49719/49719), done.
HEAD is now at c467fef3524c joplin-cli: 3.4.12 -> 3.5.1 (#459664)
Local evaluation for computing rebuilds
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/infinity/.cache/nixpkgs-review/rev-0d2ba1acef8545dc078e13afafb1a905f4d06b17/nixpkgs nixpkgs-overlays=/tmp/tmpjf640jhz -qaP --xml --out-path --show-trace --no-allow-import-from-derivation
$ git merge --no-commit --no-ff 0d2ba1acef8545dc078e13afafb1a905f4d06b17
Automatic merge went well; stopped before committing as requested
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/infinity/.cache/nixpkgs-review/rev-0d2ba1acef8545dc078e13afafb1a905f4d06b17/nixpkgs nixpkgs-overlays=/tmp/tmpjf640jhz -qaP --xml --out-path --show-trace --no-allow-import-from-derivation --meta
--------- Impacted packages on 'x86_64-linux' ---------
1 package updated:
ckb-next (0.6.2 → 0.6.2-source-2025-09-25)

warning: ignoring the client-specified setting 'system', because it is a restricted setting and you are not a trusted user

$ nix build --file /nix/store/n7wf34ag2lnvqsgq4hclh79qy55qyqcq-nixpkgs-review-3.5.1/lib/python3.13/site-packages/nixpkgs_review/nix/review-shell.nix --nix-path 'nixpkgs=/home/infinity/.cache/nixpkgs-review/rev-0d2ba1acef8545dc078e13afafb1a905f4d06b17/nixpkgs nixpkgs-overlays=/tmp/tmpjf640jhz' --extra-experimental-features 'nix-command no-url-literals' --no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed --argstr local-system x86_64-linux --argstr nixpkgs-path /home/infinity/.cache/nixpkgs-review/rev-0d2ba1acef8545dc078e13afafb1a905f4d06b17/nixpkgs --argstr nixpkgs-config-path /tmp/tmpdjvd1eay.nix --argstr attrs-path /home/infinity/.cache/nixpkgs-review/rev-0d2ba1acef8545dc078e13afafb1a905f4d06b17/attrs.nix
warning: ignoring the client-specified setting 'sandbox', because it is a restricted setting and you are not a trusted user
--------- Report for 'x86_64-linux' ---------
1 package built:
ckb-next

Logs can be found under:
/home/infinity/.cache/nixpkgs-review/rev-0d2ba1acef8545dc078e13afafb1a905f4d06b17/logs


$ /nix/store/rqn7v98b071xg19iyrz3qzlby8nwmqh3-nix-2.31.2/bin/nix-shell --argstr local-system x86_64-linux --argstr nixpkgs-path /home/infinity/.cache/nixpkgs-review/rev-0d2ba1acef8545dc078e13afafb1a905f4d06b17/nixpkgs --argstr nixpkgs-config-path /tmp/tmpdjvd1eay.nix --argstr attrs-path /home/infinity/.cache/nixpkgs-review/rev-0d2ba1acef8545dc078e13afafb1a905f4d06b17/attrs.nix --nix-path 'nixpkgs=/home/infinity/.cache/nixpkgs-review/rev-0d2ba1acef8545dc078e13afafb1a905f4d06b17/nixpkgs nixpkgs-overlays=/tmp/tmpjf640jhz' /nix/store/n7wf34ag2lnvqsgq4hclh79qy55qyqcq-nixpkgs-review-3.5.1/lib/python3.13/site-packages/nixpkgs_review/nix/review-shell.nix
```
</details>

4JX commented on pkgs/tools/misc/ckb-next/default.nix:
```diff
@@ -12,24 +12,27 @@
+  rev = "4bf942dba5e73c2778ef797b6b8dd6b0239aca9a";
+in
+stdenv.mkDerivation {
+  version = "source-${builtins.substring 0 7 rev}";
```
Fixed, don't know how I missed that.

LordGrimmauld commented on pkgs/by-name/ck/ckb-next/package.nix:
```diff
@@ -1,51 +1,48 @@
-stdenv.mkDerivation rec {
-  version = "0.6.2";
+stdenv.mkDerivation {
+  version = "0.6.2-source-2025-09-25";
```
```suggestion
  version = "0.6.2-unstable-2025-09-25";
```

4JX replied:
Whoops, fixed

LordGrimmauld reviewed (changes requested):
Looking better, but the version string is still wrong

4JX commented:
There's been a few more commits to the source repo since but I decided against updating it out of not having tested what difference they make, fyi

LordGrimmauld reviewed (approved):
Built and runs, and i don't see obvious issues anymore

4JX committed:
ckb-next: Switch to Qt6

Move to by-name, and temporarily switch to pinning from a commit to solve the build issue.

emilazy reviewed (approved):
LGTM, sorry for the regression.
